image-builder/builder-services/src/main/java/io/hyscale/builder/services/docker/impl/DockerRESTClient.java

```diff
@@ -374,13 +383,22 @@ private void handleOutput(boolean isVerbose, ActivityContext activityContext, St
         }
     }
 
+    private AuthConfigurations getAuthConfigs(Map<String, ImageRegistry> registryMap) {
+        if (registryMap == null) {
+            return null;
+        }
+        AuthConfigurations authConfigs = new AuthConfigurations();
+        registryMap.entrySet().forEach(registryEntry -> authConfigs.addConfig(getAuthConfig(registryEntry.getValue())));
+        return authConfigs;
+    }
+
     private AuthConfig getAuthConfig(ImageRegistry imageRegistry) {
         if (imageRegistry == null) {
             return null;
         }
         AuthConfig authConfig = new AuthConfig();
         authConfig.withRegistryAddress(imageRegistry.getUrl());
-        String decodedAuth = new String(Base64.getDecoder().decode(imageRegistry.getToken()));
+        String decodedAuth = EncodeDecodeUtil.decode(imageRegistry.getToken());
         int delimiter = decodedAuth.indexOf(':');
         if (delimiter > 0) {
```

Comment:
Can we simplify this into a separate method in EncodeDecodeUtil?
like getDecodedUserName()
what say?